OpenRA.Game/Orders/UnitOrderGenerator.cs

```diff
@@ -87,6 +87,20 @@ public string GetCursor(World world, CPos xy, MouseInput mi)
 			return cursorName ?? (useSelect ? "select" : "default");
 		}
 
+		// Used for classic mouse orders, determines whether or not action at xy is move or select
+		public static bool InputOverridesSelection(World world, int2 xy, MouseInput mi)
+		{
+			var tar = Target.FromActor(world.ScreenMap.ActorsAt(xy).WithHighestSelectionPriority());
+			var self = world.Selection.Actors.WithHighestSelectionPriority();
+
+			var o = OrderForUnit(self, tar, mi);
+
+			if (o == null || o.Trait is IMove)
```

Comment:
Agreed. It could fix this minor issue where transports that don't have passengers can't be individually picked out of a group. Although, unless I'm misremembering, C&C logic has selection overrides deployment.